internal/webhooks/cluster.go

```diff
@@ -362,3 +365,84 @@ func (webhook *Cluster) getClusterClassForCluster(ctx context.Context, cluster *
 	}
 	return clusterClass, nil
 }
+
+func validateMachineHealthCheckTopologies(cluster *clusterv1.Cluster, clusterClass *clusterv1.ClusterClass) field.ErrorList {
+	var allErrs field.ErrorList
+
+	// Validation ControlPlane MachineHealthCheck if defined.
+	if cluster.Spec.Topology.ControlPlane.MachineHealthCheck != nil && !cluster.Spec.Topology.ControlPlane.MachineHealthCheck.MachineHealthCheckClass.IsZero() {
+		// Ensure ControlPlane does not define a MachineHealthCheck if it does not define MachineInfrastructure.
+		if clusterClass.Spec.ControlPlane.MachineInfrastructure == nil {
+			allErrs = append(allErrs, field.Forbidden(
+				field.NewPath("spec", "topology", "controlPlane", "machineHealthCheck", "spec"),
```

Comment:
```suggestion
				field.NewPath("spec", "topology", "controlPlane", "machineHealthCheck"),
```
same in other cases below